pywhy_graphs/algorithms/pag.py

```diff
@@ -1176,3 +1187,190 @@ def pag_to_mag(graph):
         mag.add_edge(u, v, mag.directed_edge_name)
 
     return mag
+
+
+def legal_pag(G: PAG, L: Optional[set] = None, S: Optional[set] = None):
+    """Checks if the provided graph is a valid Partial ancestral graph (MAG).
+
+    A valid PAG as defined in :footcite:`Zhang2008` is a mixed edge graph that
+    has no directed or almost directed cycles and no inducing paths between
+    any two non-adjacent pair of nodes.
+
```

Comment:
Perhaps describe the n-step process of the algorithm?

1. first check if there are any cycles, almost directed cycles, etc.
2. Then what doe sit do?